src/main/java/duke/task/TimedTask.java

```diff
@@ -0,0 +1,40 @@
+package duke.task;
+
+import duke.Message;
+import duke.exception.WrongNumberOfArgumentsException;
+
+abstract class TimedTask extends Task {
+    private String dateTime;
+    private final String preposition;
+
+    TimedTask(String userInput, String preposition, Types type) throws WrongNumberOfArgumentsException {
+        //TODO: Is using constructor class to parse inputs good OOP?
+        super(type);
+        String[] inputs = userInput.split(Message.WHITESPACE_REGEX + '/' + preposition + Message.WHITESPACE_REGEX);
+        if (inputs.length != 2) {
+            throw new WrongNumberOfArgumentsException(type.toString(), preposition);
+        }
+        this.setDescription(inputs[0]);
+        this.dateTime = inputs[1];
+        this.preposition = preposition;
```

Comment:
Perhaps adding these to a separate function and using constructor for just initialising would be better. Also why not send directly the parsed input to the contructor? Basically why not handle the parsing in some other class and just pass the data in required format